pdns/dnsdist-tcp.cc

```diff
@@ -435,6 +435,9 @@ void IncomingTCPConnectionState::resetForNewQuery()
   d_buffer.resize(sizeof(uint16_t));
   d_currentPos = 0;
   d_querySize = 0;
+  d_xfrMasterSerial = 0;
+  d_xfrSerialCount = 0;
+  d_xfrMasterSerialCount = 0;
```

Comment:
I thought of that as well but then I changed my mind in case we want to reuse the connections for other users and keep track of the numbers per user?

Reply:
You are right, of course! I think the goal is to be able to have concurrent XFRs over the same incoming connection, but also concurrent XFRs over the same outgoing connection, so I guess we need to store these counters in the object representing the query instead of the one representing a connection. Let's keep it as it is for now, that's fine.